### PR TITLE
Slurm memory

### PR DESCRIPTION
**Pull request by darafferty**
Add option to specify the amount of memory required for slurm clusters.

### CODE REVIEW

File: examples/rapthor.parset [hunk 1]
```diff
@@ -142,13 +142,15 @@ input_skymodel = /data/skymodel.txt
...
-# of processors per task to request from SLURM can be specified with the
-# cpus_per_task option (default = 0 = all). By setting the cpus_per_task value to the
-# number of processors per node, one can ensure that each task gets the entire
-# node to itself, which is the recommended way of running Rapthor
+# once can be specified with the max_nodes option (default = 12), the number of
+# processors and amount of memory per node to request from SLURM can be
+# specified with the cpus_per_task (default = 0 = all) and mem_per_node_gb
+# options (default = 0 = all). By setting the cpus_per_task value to the number of
+# processors per node, one can ensure that each task gets the entire node to
+# itself, which is the recommended way of running Rapthor
```
Comment (gmloose) [inline examples/rapthor.parset:150]: This only works if you have a (more or less) homogeneous cluster. What to do if you have nodes with different numbers of CPUs?

File: rapthor/lib/operation.py [hunk 2]
```diff
@@ -106,6 +106,15 @@ def __init__(self, field, name=None, index=None):
...
         self.cpus_per_task = self.parset['cluster_specific']['cpus_per_task']
 
+        # Get the amount of memory in GB per node (SLRUM only). This is passed to
+        # sbatch's --mem option (see https://slurm.schedmd.com/sbatch.html)
+        if self.parset['cluster_specific']['mem_per_node_gb'] == 0:
+            # Handle special case in which all memory is requested (must be passed to
+            # slurm as '0', without units)
+            self.mem_per_node_gb = '0'
+        else:
+            self.mem_per_node_gb = '{}G'.format(self.parset['cluster_specific']['mem_per_node_gb'])
```
Comment (gmloose) [inline rapthor/lib/operation.py:116]: I guess this can also be a floating point number, right?

Comment (gmloose) [reply rapthor/lib/operation.py:116]: I see below that you read it in as an integer :)

**Review by gmloose — APPROVED** [review]

File: examples/rapthor.parset (hunk 1 above)
Comment (darafferty) [reply examples/rapthor.parset:150]: Hmm, good question. I expect most such clusters would have different partitions for the different node types, but specifying a partition is not supported yet in Rapthor either.

**Pull request merged**